<div align="right" style=" font-size: 80%; text-align: center; margin: 0 auto">
<img src="https://raw.githubusercontent.com/Explore-AI/Pictures/master/ExploreAI_logos/Logo blue_dark.png"  style="width:25px" align="right";/>
</div>

# Logical and comparison operators ii
© ExploreAI Academy

> ⚠️ This notebook will not run on Google Colab because it cannot connect to a local database. Please make sure that this notebook is running on the same local machine as your MySQL Workbench installation and MySQL `united_nations` database.

## Learning objectives

In this train we will learn:
- How to use the IS NULL and IS NOT NULL statements
- How to use the IS IN and IS NOT IN statements
- Understand if there is any correlation between the GDP and availibility of drinking water and sanitation services in Sub-Saharan Africa

## Connecting to our MySQL database

Using our `Access_to_Basic_Services` table in our `united_nations` database we created in MySQL Workbench, we want to answer some questions about our dataset. We can apply the same queries we used in MySQL Workbench in this notebook if we connect to our MySQL server by running the cells below.


In [1]:
pip install --user sqlalchemy ==1.4.4

Note: you may need to restart the kernel to use updated packages.


ERROR: Invalid requirement: '==1.4.4'


In [2]:
# Load and activate the SQL extension to allows us to execute SQL in a Jupyter notebook. 
# If you get an error here, make sure that mysql and pymysql is installed correctly. 

%load_ext sql

In [3]:
# Establish a connection to the local database using the '%sql' magic command,
# Replace 'password' with our connection password and `db_name` with our database name. 
# If you get an error here, please make sure the database name or password is correct.

%sql mysql+pymysql://root:Dau/2022@localhost:3306/united_nations

'Connected: root@united_nations'


To make a query, we add the `%%sql` command to the start of a cell, create one open line and then the query like below and run the cell.

In [4]:
%%sql

SELECT 
    *
FROM
    Access_to_Basic_Services
LIMIT 5;

 * mysql+pymysql://root:***@localhost:3306/united_nations
5 rows affected.


[('Central and Southern Asia', 'Central Asia', 'Kazakhstan', 2015, Decimal('94.67'), Decimal('98.00'), Decimal('17.542806'), Decimal('184.39'), Decimal('2699700.00'), Decimal('4.93')),
 ('Central and Southern Asia', 'Central Asia', 'Kazakhstan', 2016, Decimal('94.67'), Decimal('98.00'), Decimal('17.794055'), Decimal('137.28'), Decimal('2699700.00'), Decimal('4.96')),
 ('Central and Southern Asia', 'Central Asia', 'Kazakhstan', 2017, Decimal('95.00'), Decimal('98.00'), Decimal('18.037776'), Decimal('166.81'), Decimal('2699700.00'), Decimal('4.90')),
 ('Central and Southern Asia', 'Central Asia', 'Kazakhstan', 2018, Decimal('95.00'), Decimal('98.00'), Decimal('18.276452'), Decimal('179.34'), Decimal('2699700.00'), Decimal('4.85')),
 ('Central and Southern Asia', 'Central Asia', 'Kazakhstan', 2019, Decimal('95.00'), Decimal('98.00'), Decimal('18.513673'), Decimal('181.67'), Decimal('2699700.00'), Decimal('4.80'))]

## Exercise


We will be working with the `united_nations.Access_to_Basic_Services` table, which contains information about different countries, their access to basic services, and their estimated GDP.

In this exercise We would like to determine if the GDP of a country, specifically in Sub-Saharan Africa has any correlation to its access to basic services


### Task 1 
Select data from the Sub-Saharan African region during the year 2020.

In [5]:
%%sql
SELECT
    Country_name,
    Time_period,
    Pct_managed_drinking_water_services,
    Pct_managed_sanitation_services,
    Est_gdp_in_billions,
    Region
FROM
united_nations.Access_to_Basic_Services
WHERE
Region = 'Sub-Saharan Africa'
AND Time_period = 2020

 * mysql+pymysql://root:***@localhost:3306/united_nations
47 rows affected.


[('Burundi', 2020, Decimal('70.33'), Decimal('44.33'), Decimal('2.65'), 'Sub-Saharan Africa'),
 ('Djibouti', 2020, Decimal('69.00'), Decimal('56.00'), Decimal('3.18'), 'Sub-Saharan Africa'),
 ('Ethiopia', 2020, Decimal('58.00'), Decimal('11.67'), Decimal('107.66'), 'Sub-Saharan Africa'),
 ('Kenya', 2020, Decimal('67.00'), Decimal('33.67'), Decimal('100.67'), 'Sub-Saharan Africa'),
 ('Madagascar', 2020, Decimal('56.33'), Decimal('13.00'), Decimal('13.05'), 'Sub-Saharan Africa'),
 ('Malawi', 2020, Decimal('74.33'), Decimal('28.67'), Decimal('12.18'), 'Sub-Saharan Africa'),
 ('Mauritius', 2020, Decimal('100.00'), Decimal('96.00'), Decimal('11.40'), 'Sub-Saharan Africa'),
 ('Mayotte', 2020, Decimal('96.00'), Decimal('100.00'), None, 'Sub-Saharan Africa'),
 ('Mozambique', 2020, Decimal('66.67'), Decimal('40.33'), Decimal('14.03'), 'Sub-Saharan Africa'),
 ('Rwanda', 2020, Decimal('66.33'), Decimal('64.00'), Decimal('10.18'), 'Sub-Saharan Africa'),
 ('Réunion', 2020, Decimal('100.00'), Decimal('100.00'), None, 'Sub-Saharan Africa'),
 ('Somalia', 2020, Decimal('57.33'), Decimal('40.00'), Decimal('6.88'), 'Sub-Saharan Africa'),
 ('South Sudan', 2020, Decimal('48.33'), Decimal('22.33'), None, 'Sub-Saharan Africa'),
 ('Uganda', 2020, Decimal('61.00'), Decimal('21.67'), Decimal('37.60'), 'Sub-Saharan Africa'),
 ('United Republic of Tanzania', 2020, Decimal('65.00'), Decimal('34.00'), None, 'Sub-Saharan Africa'),
 ('Zambia', 2020, Decimal('66.67'), Decimal('32.67'), Decimal('18.11'), 'Sub-Saharan Africa'),
 ('Zimbabwe', 2020, Decimal('68.00'), Decimal('36.33'), Decimal('21.51'), 'Sub-Saharan Africa'),
 ('Angola', 2020, Decimal('52.33'), Decimal('47.00'), Decimal('53.62'), 'Sub-Saharan Africa'),
 ('Cameroon', 2020, Decimal('64.00'), Decimal('43.00'), Decimal('40.77'), 'Sub-Saharan Africa'),
 ('Central African Republic', 2020, Decimal('38.33'), Decimal('15.00'), Decimal('2.33'), 'Sub-Saharan Africa'),
 ('Chad', 2020, Decimal('52.67'), Decimal('18.67'), Decimal('10.72'), 'Sub-Saharan Africa'),
 ('Congo', 2020, Decimal('69.00'), Decimal('17.67'), None, 'Sub-Saharan Africa'),
 ('Democratic Republic of the Congo', 2020, Decimal('47.67'), Decimal('15.33'), None, 'Sub-Saharan Africa'),
 ('Gabon', 2020, Decimal('73.33'), Decimal('47.00'), Decimal('15.31'), 'Sub-Saharan Africa'),
 ('Sao Tome and Principe', 2020, Decimal('77.33'), Decimal('46.00'), Decimal('0.47'), 'Sub-Saharan Africa'),
 ('Botswana', 2020, Decimal('89.67'), Decimal('74.33'), Decimal('14.93'), 'Sub-Saharan Africa'),
 ('Eswatini', 2020, Decimal('76.67'), Decimal('61.33'), Decimal('3.98'), 'Sub-Saharan Africa'),
 ('Lesotho', 2020, Decimal('76.33'), Decimal('49.67'), Decimal('2.23'), 'Sub-Saharan Africa'),
 ('Namibia', 2020, Decimal('83.67'), Decimal('35.00'), Decimal('10.58'), 'Sub-Saharan Africa'),
 ('South Africa', 2020, Decimal('92.00'), Decimal('78.67'), Decimal('337.62'), 'Sub-Saharan Africa'),
 ('Benin', 2020, Decimal('65.33'), Decimal('17.33'), Decimal('15.65'), 'Sub-Saharan Africa'),
 ('Burkina Faso', 2020, Decimal('53.33'), Decimal('25.00'), Decimal('17.93'), 'Sub-Saharan Africa'),
 ('Cabo Verde', 2020, Decimal('87.33'), Decimal('78.00'), Decimal('1.70'), 'Sub-Saharan Africa'),
 ("Côte d'Ivoire", 2020, Decimal('70.67'), Decimal('34.67'), None, 'Sub-Saharan Africa'),
 ('Gambia', 2020, Decimal('79.33'), Decimal('44.33'), None, 'Sub-Saharan Africa'),
 ('Ghana', 2020, Decimal('84.67'), Decimal('23.00'), Decimal('70.04'), 'Sub-Saharan Africa'),
 ('Guinea', 2020, Decimal('67.33'), Decimal('32.33'), Decimal('14.18'), 'Sub-Saharan Africa'),
 ('Guinea-Bissau', 2020, Decimal('60.00'), Decimal('19.33'), Decimal('1.43'), 'Sub-Saharan Africa'),
 ('Liberia', 2020, Decimal('75.00'), Decimal('17.67'), Decimal('3.04'), 'Sub-Saharan Africa'),
 ('Mali', 2020, Decimal('83.67'), Decimal('46.00'), Decimal('17.47'), 'Sub-Saharan Africa'),
 ('Mauritania', 2020, Decimal('70.33'), Decimal('48.00'), Decimal('8.41'), 'Sub-Saharan Africa'),
 ('Niger', 2020, Decimal('57.33'), Decimal('24.67'), Decimal('13.

### Task 2

Sometimes there are null values in our entries. Any country having Null values for their GDP should not be included in our query as they will not help us determine if there is any correlation between GDP and access to basic services. For this task determine if there are any NULL values in the GDP column

In [6]:
%%sql
# Checking for null values---- no null values detected-----
SELECT
    Country_name,
    Time_period,
    Pct_managed_drinking_water_services,
    Pct_managed_sanitation_services,
    Est_gdp_in_billions,
    Region
FROM
united_nations.Access_to_Basic_Services
WHERE
Est_gdp_in_billions = 0

 * mysql+pymysql://root:***@localhost:3306/united_nations
0 rows affected.


[]

### Task 3

If there are any Null values exclude them from your query.

In [19]:
%%sql
# Excluding null values in my querry
SELECT
    Country_name,
    Time_period,
    Pct_managed_drinking_water_services,
    Pct_managed_sanitation_services,
    Est_gdp_in_billions,
    Region
FROM
united_nations.Access_to_Basic_Services
WHERE
Est_gdp_in_billions <> 0

    



 * mysql+pymysql://root:***@localhost:3306/united_nations
800 rows affected.


[('Kazakhstan', 2015, Decimal('94.67'), Decimal('98.00'), Decimal('184.39'), 'Central and Southern Asia'),
 ('Kazakhstan', 2016, Decimal('94.67'), Decimal('98.00'), Decimal('137.28'), 'Central and Southern Asia'),
 ('Kazakhstan', 2017, Decimal('95.00'), Decimal('98.00'), Decimal('166.81'), 'Central and Southern Asia'),
 ('Kazakhstan', 2018, Decimal('95.00'), Decimal('98.00'), Decimal('179.34'), 'Central and Southern Asia'),
 ('Kazakhstan', 2019, Decimal('95.00'), Decimal('98.00'), Decimal('181.67'), 'Central and Southern Asia'),
 ('Kazakhstan', 2020, Decimal('95.00'), Decimal('98.00'), Decimal('171.08'), 'Central and Southern Asia'),
 ('Tajikistan', 2015, Decimal('80.33'), Decimal('95.00'), Decimal('8.27'), 'Central and Southern Asia'),
 ('Tajikistan', 2016, Decimal('81.33'), Decimal('95.33'), Decimal('6.99'), 'Central and Southern Asia'),
 ('Tajikistan', 2017, Decimal('82.33'), Decimal('95.67'), Decimal('7.54'), 'Central and Southern Asia'),
 ('Tajikistan', 2018, Decimal('83.33'), Decimal('95.67'), Decimal('7.77'), 'Central and Southern Asia'),
 ('Tajikistan', 2019, Decimal('85.00'), Decimal('96.33'), Decimal('8.30'), 'Central and Southern Asia'),
 ('Tajikistan', 2020, Decimal('85.00'), Decimal('96.33'), Decimal('8.13'), 'Central and Southern Asia'),
 ('Turkmenistan', 2015, Decimal('97.67'), Decimal('97.33'), Decimal('35.80'), 'Central and Southern Asia'),
 ('Turkmenistan', 2016, Decimal('98.00'), Decimal('98.00'), Decimal('36.17'), 'Central and Southern Asia'),
 ('Turkmenistan', 2017, Decimal('98.67'), Decimal('98.00'), Decimal('37.93'), 'Central and Southern Asia'),
 ('Turkmenistan', 2018, Decimal('99.33'), Decimal('98.67'), Decimal('40.77'), 'Central and Southern Asia'),
 ('Turkmenistan', 2019, Decimal('100.00'), Decimal('99.33'), Decimal('45.23'), 'Central and Southern Asia'),
 ('Uzbekistan', 2015, Decimal('97.67'), Decimal('100.00'), Decimal('86.20'), 'Central and Southern Asia'),
 ('Uzbekistan', 2016, Decimal('97.67'), Decimal('100.00'), Decimal('86.14'), 'Central and Southern Asia'),
 ('Uzbekistan', 2017, Decimal('98.00'), Decimal('100.00'), Decimal('62.08'), 'Central and Southern Asia'),
 ('Uzbekistan', 2018, Decimal('98.00'), Decimal('100.00'), Decimal('52.63'), 'Central and Southern Asia'),
 ('Uzbekistan', 2019, Decimal('98.00'), Decimal('100.00'), Decimal('59.91'), 'Central and Southern Asia'),
 ('Uzbekistan', 2020, Decimal('98.00'), Decimal('100.00'), Decimal('59.89'), 'Central and Southern Asia'),
 ('Afghanistan', 2015, Decimal('67.00'), Decimal('45.67'), Decimal('20.00'), 'Central and Southern Asia'),
 ('Afghanistan', 2016, Decimal('69.67'), Decimal('47.00'), Decimal('18.02'), 'Central and Southern Asia'),
 ('Afghanistan', 2017, Decimal('72.33'), Decimal('49.33'), Decimal('18.90'), 'Central and Southern Asia'),
 ('Afghanistan', 2018, Decimal('75.33'), Decimal('50.67'), Decimal('18.42'), 'Central and Southern Asia'),
 ('Afghanistan', 2019, Decimal('78.00'), Decimal('52.33'), Decimal('18.90'), 'Central and Southern Asia'),
 ('Afghanistan', 2020, Decimal('80.33'), Decimal('54.00'), Decimal('20.14'), 'Central and Southern Asia'),
 ('Bangladesh', 2015, Decimal('97.33'), Decimal('47.33'), Decimal('195.08'), 'Central and Southern Asia'),
 ('Bangladesh', 2016, Decimal('97.33'), Decimal('48.67'), Decimal('265.24'), 'Central and Southern Asia'),
 ('Bangladesh', 2017, Decimal('97.33'), Decimal('50.00'), Decimal('293.75'), 'Central and Southern Asia'),
 ('Bangladesh', 2018, Decimal('97.33'), Decimal('51.33'), Decimal('321.38'), 'Central and Southern Asia'),
 ('Bangladesh', 2019, Decimal('97.67'), Decimal('52.67'), Decimal('351.24'), 'Central and Southern Asia'),
 ('Bangladesh', 2020, Decimal('97.67'), Decimal('54.00'), Decimal('373.90'), 'Central and Southern Asia'),
 ('Bhutan', 2015, Decimal('96.33'), Decimal('70.67'), Decimal('2.00'), 'Central and Southern Asia'),
 ('Bhutan', 2016, Decimal('97.00'), Decimal('71.67'), Decimal('2.16'), 'Central and Southern Asia'),
 ('Bhutan', 2017, Decimal('97.33'), Decimal('73.33'), Decim

### Task 4

Lets get an idea if there's any correlation between the GDP and access to basic services for the top 5 economies in Sub-Saharan Africr. The top 5 GDP's are : ('Nigeria','South Africa','Ethiopia','Kenya','Ghana'). Make sure your query only includes these countries.

In [23]:
%%sql

SELECT
    Country_name,
    Time_period,
    Pct_managed_drinking_water_services,
    Pct_managed_sanitation_services,
    Est_population_in_millions,
    Est_gdp_in_billions
FROM
    Access_to_Basic_Services
WHERE
    Region = 'Sub-Saharan Africa'
AND
    Time_period = 2020
AND
   Est_gdp_in_billions IS NOT NULL
AND
    Country_name IN ('Nigeria','South Africa','Ethiopia','Kenya','Ghana')
ORDER BY Est_gdp_in_billions
LIMIT 10;

 * mysql+pymysql://root:***@localhost:3306/united_nations
5 rows affected.


[('Ghana', 2020, Decimal('84.67'), Decimal('23.00'), Decimal('32.180401'), Decimal('70.04')),
 ('Kenya', 2020, Decimal('67.00'), Decimal('33.67'), Decimal('51.985780'), Decimal('100.67')),
 ('Ethiopia', 2020, Decimal('58.00'), Decimal('11.67'), Decimal('117.190911'), Decimal('107.66')),
 ('South Africa', 2020, Decimal('92.00'), Decimal('78.67'), Decimal('58.801927'), Decimal('337.62')),
 ('Nigeria', 2020, Decimal('77.33'), Decimal('42.67'), Decimal('208.327405'), Decimal('432.20'))]

Based on your results. Is there any correlation between the GDP and access to basic services? Does a higher GDP translate to better services?

### Task 5

We only looked at 5 countries in the previous query. Lets have a look at the rest of Sub-Saharan Africa .
Exclude the countries mentioned in the previous task

In [24]:
%%sql

SELECT 
    Country_name,
    Time_period,
    Pct_managed_drinking_water_services,
    Pct_managed_sanitation_services,
    Est_population_in_millions,
    Est_gdp_in_billions
FROM 
    Access_to_Basic_Services

WHERE
    Region = 'Sub-Saharan Africa'
AND 
    Time_period = 2020

AND 
    Est_gdp_in_billions IS NOT NULL
AND
    Country_name NOT IN ('Nigeria','South Africa','Ethiopia','Kenya','Ghana')

ORDER BY Est_gdp_in_billions;

 * mysql+pymysql://root:***@localhost:3306/united_nations
33 rows affected.


[('Sao Tome and Principe', 2020, Decimal('77.33'), Decimal('46.00'), Decimal('0.218641'), Decimal('0.47')),
 ('Guinea-Bissau', 2020, Decimal('60.00'), Decimal('19.33'), Decimal('2.015828'), Decimal('1.43')),
 ('Cabo Verde', 2020, Decimal('87.33'), Decimal('78.00'), Decimal('0.582640'), Decimal('1.70')),
 ('Lesotho', 2020, Decimal('76.33'), Decimal('49.67'), Decimal('2.254100'), Decimal('2.23')),
 ('Central African Republic', 2020, Decimal('38.33'), Decimal('15.00'), Decimal('5.343020'), Decimal('2.33')),
 ('Burundi', 2020, Decimal('70.33'), Decimal('44.33'), Decimal('12.220227'), Decimal('2.65')),
 ('Liberia', 2020, Decimal('75.00'), Decimal('17.67'), Decimal('5.087584'), Decimal('3.04')),
 ('Djibouti', 2020, Decimal('69.00'), Decimal('56.00'), Decimal('1.090156'), Decimal('3.18')),
 ('Eswatini', 2020, Decimal('76.67'), Decimal('61.33'), Decimal('1.180655'), Decimal('3.98')),
 ('Sierra Leone', 2020, Decimal('65.00'), Decimal('17.33'), Decimal('8.233970'), Decimal('4.06')),
 ('Somalia', 2020, Decimal('57.33'), Decimal('40.00'), Decimal('16.537016'), Decimal('6.88')),
 ('Togo', 2020, Decimal('70.67'), Decimal('20.00'), Decimal('8.442580'), Decimal('7.57')),
 ('Mauritania', 2020, Decimal('70.33'), Decimal('48.00'), Decimal('4.498604'), Decimal('8.41')),
 ('Rwanda', 2020, Decimal('66.33'), Decimal('64.00'), Decimal('13.146362'), Decimal('10.18')),
 ('Namibia', 2020, Decimal('83.67'), Decimal('35.00'), Decimal('2.489098'), Decimal('10.58')),
 ('Chad', 2020, Decimal('52.67'), Decimal('18.67'), Decimal('16.644701'), Decimal('10.72')),
 ('Mauritius', 2020, Decimal('100.00'), Decimal('96.00'), Decimal('1.265740'), Decimal('11.40')),
 ('Malawi', 2020, Decimal('74.33'), Decimal('28.67'), Decimal('19.377061'), Decimal('12.18')),
 ('Madagascar', 2020, Decimal('56.33'), Decimal('13.00'), Decimal('28.225177'), Decimal('13.05')),
 ('Niger', 2020, Decimal('57.33'), Decimal('24.67'), Decimal('24.333639'), Decimal('13.74')),
 ('Mozambique', 2020, Decimal('66.67'), Decimal('40.33'), Decimal('31.178239'), Decimal('14.03')),
 ('Guinea', 2020, Decimal('67.33'), Decimal('32.33'), Decimal('13.205153'), Decimal('14.18')),
 ('Botswana', 2020, Decimal('89.67'), Decimal('74.33'), Decimal('2.546402'), Decimal('14.93')),
 ('Gabon', 2020, Decimal('73.33'), Decimal('47.00'), Decimal('2.292573'), Decimal('15.31')),
 ('Benin', 2020, Decimal('65.33'), Decimal('17.33'), Decimal('12.643123'), Decimal('15.65')),
 ('Mali', 2020, Decimal('83.67'), Decimal('46.00'), Decimal('21.224040'), Decimal('17.47')),
 ('Burkina Faso', 2020, Decimal('53.33'), Decimal('25.00'), Decimal('21.522626'), Decimal('17.93')),
 ('Zambia', 2020, Decimal('66.67'), Decimal('32.67'), Decimal('18.927715'), Decimal('18.11')),
 ('Zimbabwe', 2020, Decimal('68.00'), Decimal('36.33'), Decimal('15.669666'), Decimal('21.51')),
 ('Senegal', 2020, Decimal('85.00'), Decimal('57.00'), Decimal('16.436120'), Decimal('24.49')),
 ('Uganda', 2020, Decimal('61.00'), Decimal('21.67'), Decimal('44.404611'), Decimal('37.60')),
 ('Cameroon', 2020, Decimal('64.00'), Decimal('43.00'), Decimal('26.491087'), Decimal('40.77')),
 ('Angola', 2020, Decimal('52.33'), Decimal('47.00'), Decimal('33.428486'), Decimal('53.62'))]

Again. Do you see any correlation? Arrange the `Pct_managed_drinking_water_services` in descending order. Do the countries with the highest percentage to `Pct_managed_drinking_water_services` have a higher GDP then the countries with a lower percentage of `Pct_managed_drinking_water_services` ?
 Now arrange the `Pct_managed_sanitation_services` in descending order. Do countries with a higher `Pct_managed_sanitation_services` also have a higher GDP?

## Solutions

### Task 1

In [ ]:
%%sql

SELECT 
	Country_name,
	Time_period,
	Pct_managed_drinking_water_services,
	Pct_managed_sanitation_services,
	Est_population_in_millions,
	Est_gdp_in_billions
FROM 
	united_nations.Access_to_Basic_Services
WHERE
	region = ‘Sub-Saharan Africa’
AND 
    Time_period = 2020;

### Task 2

In [ ]:
%%sql

SELECT 
	Country_name,
	Time_period,
	Pct_managed_drinking_water_services,
	Pct_managed_sanitation_services,
	Est_gdp_in_billions,
    region
FROM 
	united_nations.Access_to_Basic_Services
WHERE
	region = 'Sub-Saharan Africa'
AND 
	Time_period = 2020
AND 
	Est_gdp_in_billions IS NULL;

# Use LIMIT if you think the results set will be large



### Task 3

In [ ]:
%%sql

SELECT 
	Country_name,
	Time_period,
	Pct_managed_drinking_water_services,
	Pct_managed_sanitation_services,
	Est_gdp_in_billions,
    	region
FROM 
	united_nations.Access_to_Basic_Services
WHERE
	region = 'Sub-Saharan Africa'
AND 
	Time_period = 2020
AND 
	Est_gdp_in_billions IS NOT NULL;

# Use LIMIT if you think the results set will be large
# Use ORDER BY Est_gdp_in_billions to order your results

### Task 4

Arrange the data in descending order. Comparing Nigeria to South Africa.  We see that Nigeria has a lower percentage in the availability of both water and sanitation services. 


In [ ]:
%%sql

SELECT 
	Country_name,
	Time_period,
	Pct_managed_drinking_water_services,
	Pct_managed_sanitation_services,
	Est_population_in_millions,
	Est_gdp_in_billions
FROM 
	united_nations.Access_to_Basic_Services

WHERE
	Region = ‘Sub-Saharan Africa
AND 
Time_period = 2020

AND 
	Est_gdp_in_billions IS NOT NULL

AND 
	Country_name  NOT IN ('Nigeria','South Africa','Ethiopia','Kenya','Ghana')


# Use LIMIT if you think the results set will be large
# Use ORDER BY Est_gdp_in_billions to order your results


### Task 5

Looking at the first entry without ordering any data it is interesting to note that Burundi, with a GDP of only 2.65 Billion has a similar percentage of managed services compared to Nigeria seen in our previous  query(77% and 42%). Nigeria had a percentage of 77 for managed water and 42 for managed sanitation. Arranging the percentage managed drinking water services in descending order.Mauritius with a GDP lower than that of Botswana has better drinking water and sanitation services. 

In [ ]:
%%sql

SELECT 
	Country_name,
	Time_period,
	Pct_managed_drinking_water_services,
	Pct_managed_sanitation_services,
	Est_population_in_millions,
	Est_gdp_in_billions
FROM 
	united_nations.Access_to_Basic_Services

WHERE
	Region = ‘Sub-Saharan Africa
AND 
Time_period = 2020

AND 
	Est_gdp_in_billions IS NOT NULL

AND 
	Country_name  IS IN ('Nigeria','South Africa','Ethiopia','Kenya','Ghana');

# Use LIMIT if you think the results set will be large
# Use ORDER BY Est_gdp_in_billions to order your results by GDP
# Use ORDER BY Pct_managed_drinking_water_services to order percentage managed drinking water services
# Use ORDER BY Pct_managed_sanitation_services to order your results by percentage managed sanitation services


## Summary

In this exercise we used the IS NULL statement to determine if there were any null values in the GDP column. 
We then used the IS NOT NULL statement to exclude those nulls.
We looked at the top 5 GDP's in Sub-Saharan Africa by using the IS IN statement. 
We had a look at the rest of the Sub-Saharan African countries by excluding the top 5 GDP's using the IS NOT IN statement
Based on the quick perusal of the data in Sub-Saharan africa, we can conclude that there isn’t
any noticeable correlation between GDP and the availability of drinking water and sanitation services. 


#  

<div align="center" style=" font-size: 80%; text-align: center; margin: 0 auto">
<img src="https://raw.githubusercontent.com/Explore-AI/Pictures/master/ExploreAI_logos/EAI_Blue_Dark.png"  style="width:200px";/>
</div>